src/Disks/HDFS/DiskHDFS.cpp

```diff
@@ -57,9 +57,11 @@ DiskHDFS::DiskHDFS(
     const String & hdfs_root_path_,
     SettingsPtr settings_,
     const String & metadata_path_,
-    const Poco::Util::AbstractConfiguration & config_)
+    const Poco::Util::AbstractConfiguration & config_,
+    const Settings & contextSettings_)
     : IDiskRemote(disk_name_, hdfs_root_path_, metadata_path_, "DiskHDFS", settings_->thread_pool_size)
     , config(config_)
+    , contextSettings(contextSettings_)
```

Comment:
I would rather put the value `hdfs_replication` from context settings to disk settings (in getSettings function below), so there will be no need to save context settings in disk.

Reply:
done. Could you help continue review it？ Thank you.